libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -887,15 +921,36 @@ namespace pika::mpi::experimental {
         }
 
         // clean up if we initialized mpi
+        PIKA_DETAIL_DP(detail::mpi_debug<1>, debug(str<>("finalize"), detail::mpi_data_));
         mpi::detail::environment::finalize();
+    }
 
-        PIKA_DETAIL_DP(detail::mpi_debug<5>,
-            debug(str<>("Clearing mode"), detail::mpi_data_, "disable_user_polling"));
+    // -----------------------------------------------------------------
+    size_t get_work_count() { return detail::mpi_data_.all_in_flight_; }
 
-        if (pool_name.empty())
+    // -----------------------------------------------------------------
+    bool get_enable_pool() { return detail::enable_pool_; }
+    void set_enable_pool(bool mode) { detail::enable_pool_ = mode; }
```

Comment:
Naming and namespace:
- `get_enable_pool` -> `pool_enabled` or something like that?
- at least `set_enable_pool` into `detail`, if not both? They're only used internally as far as I can tell right now. We can always make them public later if needed.

I feel like the `set/get_enable_pool` leaks the internal details in the sense that you're just setting/getting an internal variable. If they move to `detail` I think the names are fine as they won't be user-facing anymore then.